core/src/test/java/org/apache/gravitino/credential/TestCredentialCacheKey.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.credential;
+
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableSet;
+
+public class TestCredentialCacheKey {
+
+  @Test
+  void testCredentialCacheKey() {
+
+    PathBasedCredentialContext context1 =
+        new PathBasedCredentialContext("user1", ImmutableSet.of("path1"), ImmutableSet.of("path2"));
+    PathBasedCredentialContext context2 =
+        new PathBasedCredentialContext("user2", ImmutableSet.of("path1"), ImmutableSet.of("path2"));
+    PathBasedCredentialContext context3 =
+        new PathBasedCredentialContext("user3", ImmutableSet.of("path3"), ImmutableSet.of("path4"));
```

Comment:
Would it be better to set `user1` in context3 to test only different path?

Would it be better to name them `context`, `contextWithDiffUser`, and `contextWithDiffPath`?
This approach seems much clearer and eliminates the need for comments.
However, I’m fine with the current approach.

Reply:
updated